migrations/Qi.sql

```diff
@@ -1,3 +1,143 @@
 -- 1 up
 
--- 1 down
\ No newline at end of file
+CREATE TABLE if not exists asset_types (
+  asset_type_id int NOT NULL primary key,
+  name varchar(255) DEFAULT NULL
+);
+
+CREATE TABLE if not exists assets (
+  tag varchar(7) NOT NULL DEFAULT '' primary key,
+  parenttag varchar(7) DEFAULT NULL,
+  customer_tag varchar(32) DEFAULT NULL,
+  customer_id int DEFAULT NULL,
+  received date DEFAULT NULL,
+  serial_number varchar(64) DEFAULT NULL,
+  asset_type_id int DEFAULT NULL,
+  manufacturer varchar(255) DEFAULT NULL,
+  product varchar(255) DEFAULT NULL,
+  model varchar(255) DEFAULT NULL,
+  cond_id int DEFAULT NULL,
+  location_id int DEFAULT NULL,
+  qty int DEFAULT '1',
+  status_id int DEFAULT NULL,
+  hipaa date DEFAULT NULL,
+  hipaa_person int DEFAULT NULL,
+  sold_via_id int DEFAULT NULL,
+  buyer_id int DEFAULT NULL,
+  sold_to varchar(255) DEFAULT NULL,
+  po_number varchar(32) DEFAULT NULL,
+  listed date DEFAULT NULL,
+  scrapped date DEFAULT NULL,
+  sold date DEFAULT NULL,
+  billed date DEFAULT NULL,
+  paid date DEFAULT NULL,
+  customer_paid date DEFAULT NULL,
+  shipped date DEFAULT NULL,
+  scrap_value varchar(10) DEFAULT NULL,
+  price varchar(10) DEFAULT NULL,
+  related_expenses varchar(10) DEFAULT NULL,
+  revenue_percentage varchar(10) DEFAULT NULL,
+  comments varchar(512) DEFAULT NULL,
+  change_stamp timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  add_stamp timestamp NOT NULL
+);
+
+CREATE TABLE if not exists barcode_map (
+  id int NOT NULL,
+  map varchar(255) DEFAULT NULL,
+  comments varchar(255) DEFAULT NULL,
+  PRIMARY KEY ("id")
+);
+
+CREATE TABLE if not exists buyers (
+  buyer_id int NOT NULL DEFAULT '0',
+  name varchar(255) DEFAULT NULL,
+  PRIMARY KEY ("buyer_id")
+);
+
+CREATE TABLE if not exists conds (
+  cond_id int NOT NULL,
+  name varchar(255) DEFAULT NULL,
+  PRIMARY KEY ("cond_id")
+);
+
+CREATE TABLE if not exists customers (
+  customer_id int NOT NULL,
+  name varchar(255) DEFAULT NULL,
+  revenue_percentage varchar(10) DEFAULT NULL,
+  PRIMARY KEY ("customer_id")
+);
+
+CREATE TABLE if not exists locations (
+  location_id int NOT NULL,
+  name varchar(255) DEFAULT NULL,
+  label varchar(16) DEFAULT NULL,
+  PRIMARY KEY ("location_id")
+);
+
+CREATE TABLE if not exists log (
+  id int NOT NULL,
+  tag varchar(32) DEFAULT NULL,
+  who varchar(255) DEFAULT NULL,
+  field varchar(255) DEFAULT NULL,
+  previous varchar(255) DEFAULT NULL,
+  value varchar(255) DEFAULT NULL,
+  stamp timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  PRIMARY KEY ("id")
+);
+
+CREATE TABLE if not exists sessions (
+  id char(32) NOT NULL,
+  a_session text NOT NULL,
+  PRIMARY KEY ("id")
+);
+
+CREATE TABLE if not exists sold_via (
+  sold_via_id int NOT NULL DEFAULT '0',
+  name varchar(255) DEFAULT NULL,
+  PRIMARY KEY ("sold_via_id")
+);
+
+CREATE TABLE if not exists status (
+  status_id int NOT NULL,
+  name varchar(255) DEFAULT NULL,
+  PRIMARY KEY ("status_id")
+);
+
+CREATE TABLE if not exists users (
```

Comment:
This appears to be lacking a `primary key`.  I don't know for a fact, but it seems like it should have one?